lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -684,6 +690,20 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         return false;
     }
 
+    private boolean checkFunctionPermission(
+            ConnectorSecurityContext context,
+            FunctionKind functionKind,
+            SchemaRoutineName function)
+    {
+        ConnectorIdentity identity = context.getIdentity();
+        for (FunctionAccessControlRule rule : functionRules) {
+            if (rule.matches(identity.getUser(), identity.getEnabledSystemRoles(), identity.getGroups(), functionKind, function)) {
+                return rule.canExecuteFunction();
+            }
+        }
+        return false;
```

Comment:
`false` will be returned if `functionRules` is empty, and according to 
```
    public void checkCanExecuteFunction(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName function)
    {
        if (functionKind == TABLE) { // <- removed
        if (!checkFunctionPermission(context, functionKind, function)) {
            denyExecuteFunction(function.toString());
        }
    }
```
this will return false for `function kind` != `TABLE`